File: scripts/app_transaction.js
// Copyright (C) 2018  David Capello
//
// This file is released under the terms of the MIT license.
// Read LICENSE.txt for more information.

var s = new Sprite(16, 32)
console.assert(s.width == 16)
console.assert(s.height == 32)

s.width = 20
console.assert(s.width == 20)
console.assert(s.height == 32)

s.height = 40
console.assert(s.width == 20)
console.assert(s.height == 40)

app.undo()
console.assert(s.width == 20)
console.assert(s.height == 32)

app.undo()
console.assert(s.width == 16)
console.assert(s.height == 32)

app.transaction(
    function() {
        s.width = 20
        s.height = 40
    })
console.assert(s.width == 20)
console.assert(s.height == 40)

app.undo()
console.assert(s.width == 16)
console.assert(s.height == 32)

app.redo()
console.assert(s.width == 20)
console.assert(s.height == 40)

File: scripts/app_transaction.lua
-- Copyright (C) 2018  David Capello
--
-- This file is released under the terms of the MIT license.
-- Read LICENSE.txt for more information.

local s = Sprite(16, 32)
assert(s.width == 16)
assert(s.height == 32)

s.width = 20
assert(s.width == 20)
assert(s.height == 32)

s.height = 40
assert(s.width == 20)
assert(s.height == 40)

app.undo()
assert(s.width == 20)
assert(s.height == 32)

app.undo()
assert(s.width == 16)
assert(s.height == 32)

app.transaction(
  function()
    s.width = 20
    s.height = 40
  end)
assert(s.width == 20)
assert(s.height == 40)

app.undo()
assert(s.width == 16)
assert(s.height == 32)

app.redo()
assert(s.width == 20)
assert(s.height == 40)
